net/pdns/Makefile

```diff
@@ -0,0 +1,243 @@
+  DEPENDS:=+libatomic
+  URL:=https://www.powerdns.com/auth.html
+endef
+
+define Package/pdns/description
+  PowerDNS is a versatile nameserver which supports a large number
+  of different backends ranging from simple zonefiles to relational
+  databases and load balancing/failover algorithms.
+  PowerDNS tries to emphasize speed and security.
+
+  This is the authoritative nameserver that answers questions about
+  domains that it knows about. You also need at least one backend installed to
+  serve data.
+endef
+
+PDNS_BACKENDS =
+
+define Package/pdns-backends
+$(call Package/pdns/Default)
+  TITLE+= (all backends)
+  DEPENDS+= $(PDNS_DEPENDS)
+  HIDDEN:=1
+endef
+
+define Package/pdns-backends/description
+ This meta package contains only dependencies for PowerDNS backends.
+endef
+
+# Create a meta-package of dependent backends (for ALL)
+define Package/pdns-backends/install
+  true
+endef
+
+define Package/pdns-tools
+  $(call Package/pdns/Default)
+  TITLE:=Tools for DNS debugging by PowerDNS
+  DEPENDS+=+boost +boost-program_options +libopenssl +p11-kit +protobuf
+endef
+
+define Package/pdns-tools/description
+  PowerDNS is a versatile nameserver which supports a large number
+  of different backends ranging from simple zonefiles to relational
+  databases and load balancing/failover algorithms.
+  PowerDNS tries to emphasize speed and security.
+
+  This package contains several tools to debug DNS issues. These tools do not
+  require any part of the PowerDNS server components to work.
+
+    * dnsbulktest: A resolver stress-tester
+    * dnsgram: Show per 5-second statistics to study intermittent resolver issues
+    * dnsreplay: Replay a pcap with DNS queries
+    * dnsscan: Prints the query-type amounts in a pcap
+    * dnsscope: Calculates statistics without replaying traffic
+    * dnstcpbench: Perform TCP benchmarking of DNS servers
+    * dnswasher: Clean a pcap of identifying IP information
+    * ixplore: Explore diffs from IXFRs
+    * nsec3dig: Calculate the correctness of NSEC3 proofs
+    * saxfr: AXFR zones and show extra information
+endef
+
+define Package/pdns-ixfrdist
+  $(call Package/pdns/Default)
+  TITLE:=Redistribute zones via AXFR and IXFR
+  DEPENDS+=+boost +boost-program_options +libopenssl +libyaml-cpp +p11-kit +protobuf
+endef
+
+define Package/pdns-ixfrdist/description
+  PowerDNS is a versatile nameserver which supports a large number
+  of different backends ranging from simple zonefiles to relational
+  databases and load balancing/failover algorithms.
+  PowerDNS tries to emphasize speed and security.
+
+  This package contains the ixfrdist tool that can redistribute multiple versions
+  of zones over AXFR and IXFR. This tool does not require any part of the
+  PowerDNS server components to work.
+endef
+
+define Package/pdns
+  $(call Package/pdns/Default)
+  DEPENDS+=+boost +boost-program_options +libcurl +liblua +libopenssl +libsodium +libsqlite3 +lua +p11-kit
```

Comment:
Do you really need to pull in libsodium and lua?

Reply:
I'll have a look. It might not need it.

Reply:
liblua and libsodium are required, lua is not, i've removed it.

liblua is used for LUA records, and libsodium is used to enable DNSSEC ed25519 (alg 15) support. I'm not too sure if this could be made available through libopenssl instead.

Reply:
OK, I've removed libsodium as a dependency now since the version of OpenSSL supports ed25519 out of the box